test/src/toga_test/app.py

```diff
@@ -0,0 +1,27 @@
+import pytest
+import tests
+
+import toga
+
+
+class TogaTest(toga.App):
+    def startup(self):
+        # For the "app" fixture.
+        global app
+        app = self
+
+        self.main_box = toga.Box()
+
+        self.main_window = toga.MainWindow(title=self.formal_name)
+        self.main_window.content = self.main_box
+        self.main_window.show()
+
+        self.add_background_task(test)
```

Comment:
Not sure how well this approach will work in the general case - `test()` is a blocking method, so that means the main event loop won't be actively running while the tests are running. That will definitely interfere with any display updates that are the result of tests.

Reply:
See [comment above](https://github.com/beeware/toga/pull/1687#issuecomment-1315914548): this is the next thing I'll look at.